docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -294,254 +296,289 @@ TODO
+requirements and is straightforward to implement.
+
+Before modifying the dissemination logic, we need to refactor the current implementation and the
+`Mempool` interface to clearly separate the broadcast goroutine in the mempool reactor from
+`CListMempool` that includes the mempool data structures. `CListMempool` provides two methods used
+by the broadcast code, `TxsWaitChan() <-chan struct{}` and `TxsFront() *clist.CElement`, which are
+just wrappers around the methods `WaitChan` and `Front` of the `CList` implementation. In
+particular, `TxsFront` is leaking implementation details outside the `Mempool` interface.
+
+### Checking received transactions
+
+Transactions are transmitted without lane information because peers cannot be trusted to send the
+correct data. A node may take advantage of the network by sending lower-priority transactions before
+higher-priority ones. Although the receiving node could easily verify the priority of a transaction
+when it calls `CheckTx`, it cannot detect if a peer is sending transactions out of order over a
+single P2P channel. For the moment, we leave out of the MVP any mechanism to detect and possibly
+punish nodes for this kind of behaviour.
+
+## Alternative designs
+
+### One CList for all lanes
+
+We briefly considered sharing one CList for all lanes, changing the internal logic of CList to
+accommodate lane requirements. However, this design significantly increases code complexity,
+particularly in the transaction dissemination logic.
+
+### One P2P channel per lane
+
+Since P2P channels already have a built-in priority mechanism, they present a reasonable option to
+implement transaction dissemination from lanes. By assigning a P2P channel to each lane, we could
+simply append new transactions to their respective channels and allow the P2P layer to manage the
+order of transmission. We decided against this option mainly because the prioritization logic cannot
+be easily modified without altering the P2P code, potentially affecting other non-mempool channels.
+
+Another drawback is that this option imposes a limit to the number of P2P channels. Channels use a
+byte as an ID, and the current distribution among all reactors goes up to channel `0x61`. For
+example, the current mempool’s P2P channel ID is `0x30`, which would serve as the default lane. We
+could reserve a range of channels for the mempool, such as starting from channel ID `0x80` and above
+(all channels with the most significant byte set to 1). This would provide a maximum of 128 lanes,
+which should suffice for most users.
+
+Nodes would also need to agree on the channel assignments during the P2P handshake. Currently, one
+of the conditions for the handshake to succeed is that there must exist an intersection of P2P
+channels. Since lanes are defined in the application logic, the nodes only need to agree on the
+application version, as it already happens in the current implementation.
+
+### Duality lane/priority
```

Comment:
To confirm, now lineID == priority? And the higher the greater?

Reply:
That's right.

Reply:
I assume this is written somewhere...